daemon/src/model/cfd.rs

```diff
@@ -2373,6 +2390,219 @@ mod tests {
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let _ = assert!(cfd.start_rollover().is_ok(), "Could not start rollover");
+
+        let cfd = cfd.dummy_start_collab_settlement();
+
+        let no_rollover_reason = cfd
+            .roll_over(RolloverCompleted::succeeded(
+                OrderId::default(),
+                Dlc::dummy(None),
+                FundingFee::dummy(),
+            ))
+            .unwrap_err();
+        assert_eq!(
+            no_rollover_reason,
+            NoRolloverReason::InCollaborativeSettlement
+        );
+    }
+
+    #[test]
+    fn given_ongoing_collab_settlement_then_cannot_start_rollover() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+
+        let cfd = Cfd::taker_long()
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let _ = assert!(cfd.start_rollover().is_ok(), "Could not start rollover");
+
+        let cfd = cfd.dummy_start_collab_settlement();
+
+        let err = cfd.start_rollover().unwrap_err();
+        let no_rollover_reason = err.downcast::<NoRolloverReason>().unwrap();
+        assert_eq!(
+            no_rollover_reason,
+            NoRolloverReason::InCollaborativeSettlement
+        );
+
+        let cfd = Cfd::maker_short()
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let _ = assert!(cfd.start_rollover().is_ok(), "Could not start rollover");
+
+        let cfd = cfd.dummy_start_collab_settlement();
+
+        let err = cfd.start_rollover().unwrap_err();
+        let no_rollover_reason = err.downcast::<NoRolloverReason>().unwrap();
+        assert_eq!(
+            no_rollover_reason,
+            NoRolloverReason::InCollaborativeSettlement
+        );
+    }
+
+    #[test]
+    fn given_ongoing_rollover_then_can_start_collaborative_settlement() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+        let order_id = OrderId::default();
+
+        let taker_keys = crate::keypair::new(&mut rand::thread_rng());
+        let maker_keys = crate::keypair::new(&mut rand::thread_rng());
+
+        let taker_long = Cfd::taker_long()
+            .with_id(order_id)
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let (taker_long, proposal, taker_sig, _) = taker_long
+            .with_lock(taker_keys, maker_keys)
+            .dummy_start_rollover()
+            .dummy_collab_settlement_taker(opening_price);
```

Comment:
It is pretty confusing that the "act" part of this test is chained together with the "arrange". Can we split this please?